app/screens/scheduled_post_options/scheduled_post_picker.tsx

```diff
@@ -0,0 +1,148 @@
+import BottomSheet from '@screens/bottom_sheet';
+import {dismissBottomSheet} from '@screens/navigation';
+import ScheduledPostCoreOptions from '@screens/scheduled_post_options/core_options';
+import ScheduledPostFooter from '@screens/scheduled_post_options/footer';
+import {FOOTER_HEIGHT} from '@screens/scheduled_post_options/footer/scheduled_post_footer';
+import {logDebug} from '@utils/log';
+import {showScheduledPostCreationErrorSnackbar} from '@utils/snack_bar';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {getTimezone} from '@utils/user';
+
+import type {BottomSheetFooterProps} from '@gorhom/bottom-sheet';
+
+const OPTIONS_PADDING = 12;
+const OPTIONS_SEPARATOR_HEIGHT = 1;
+const TITLE_HEIGHT = 54;
+const ITEM_HEIGHT = 48;
+
+export const SCHEDULED_POST_OPTIONS_BUTTON = 'close-scheduled-post-options';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        backgroundColor: theme.centerChannelBg,
+        height: 200,
+    },
+    titleContainer: {
+        alignItems: 'center',
+        flexDirection: 'row',
+    },
+    title: {
+        color: theme.centerChannelColor,
+        ...typography('Heading', 600, 'SemiBold'),
+    },
+    optionsContainer: {
+        paddingTop: OPTIONS_PADDING,
+    },
+    optionsSeparator: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        height: OPTIONS_SEPARATOR_HEIGHT,
+    },
+}));
+
+type Props = {
+    currentUserTimezone?: UserTimezone | null;
+    onSchedule: (schedulingInfo: SchedulingInfo) => Promise<void | {data?: boolean; error?: unknown}>;
+}
+
+export function ScheduledPostOptions({currentUserTimezone, onSchedule}: Props) {
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+    const [isScheduling, setIsScheduling] = useState(false);
+    const [selectedTime, setSelectedTime] = useState<string | null>(null);
+    const [customTimeSelected, setCustomTimeSelected] = useState(false);
+    const userTimezone = getTimezone(currentUserTimezone);
+
+    const style = getStyleSheet(theme);
+
+    const snapPoints = useMemo(() => {
+        const bottomSheetAdjust = Platform.select({ios: 5, default: 20});
+
+        // 9 items to display inline date-time picker, 4 items otherwise
+        const iosNumberOfItems = customTimeSelected ? 9 : 4;
+        const andriodNumberOfItems = customTimeSelected ? 4 : 3;
+        const numberOfItems = Platform.select({ios: iosNumberOfItems, default: andriodNumberOfItems});
+        const COMPONENT_HEIGHT = TITLE_HEIGHT + (numberOfItems * ITEM_HEIGHT) + FOOTER_HEIGHT + bottomSheetAdjust;
+        return [1, COMPONENT_HEIGHT];
+    }, [customTimeSelected]);
+
+    const onSelectTime = useCallback((selectedValue: string) => {
+        setSelectedTime(selectedValue);
+    }, []);
+
+    const handleOnSchedule = usePreventDoubleTap(useCallback(async () => {
+        if (!selectedTime) {
```

Comment:
is there a reason to log this but not to display anything to the user ?

Reply:
Added the isError state to show the error of not selecting time.